src/bit_vec.rs

```diff
@@ -68,72 +30,75 @@ impl<O: BitOrder, T: BitStore + ToByteSlice> Encode for BitVec<O, T> {
 		);
 		Compact(len as u32).encode_to(dest);
 
-		let byte_slice: &[u8] = self.as_slice().as_byte_slice();
+		// NOTE: doc of `BitSlice::as_slice`:
+		// > The returned slice handle views all elements touched by self
+		//
+		// Thus we are sure the slice doesn't contain unused elements at the end.
+		let slice = self.as_slice();
```

Comment:
Here we could also check that len is less than `ARCH32BIT_BITSLICE_MAX_BITS` otherwise it won't be able to decode.

But between panicking and having an invalid data. I think it is fine to have an invalid data. Anyway having such a big BitSlice is likely to be an error by itself.